frappe/model/base_document.py

```diff
@@ -349,6 +353,8 @@ def get_valid_dict(
 
 			if ignore_nulls and d[fieldname] is None:
 				del d[fieldname]
+			if not is_virtual_field and field_value is _DOC_DELETED_ATTR:
+				del d[fieldname]
```

Comment:
Separated because 4 conditions in 1 `if` was a bit hard to understand for me 😓 

---

Also, now that I look at this it should've been `elif` and not just `if`